TimeTracking/pages/config_update.php

```diff
@@ -21,21 +21,27 @@
    http://www.mantisbt.org/forums/viewtopic.php?f=4&t=589	
 */
 
-form_security_validate( 'plugin_TimeTracking_config_update' );
+$fid = 'plugin_TimeTracking_config_update';
+form_security_validate( $fid );
 
 access_ensure_global_level( config_get( 'manage_plugin_threshold' ) );
 
-function maybe_set_option( $name, $value ) {
-	if ( $value != plugin_config_get( $name ) ) {
-		plugin_config_set( $name, $value );
-	}
-}
 
-maybe_set_option( 'edit_threshold', gpc_get_int( 'edit_threshold' ) );
-maybe_set_option( 'view_threshold', gpc_get_int( 'view_threshold' ) );
-maybe_set_option( 'reporting_threshold', gpc_get_int( 'reporting_threshold' ) );
-maybe_set_option( 'categories', gpc_get_string( 'categories' ) );
-maybe_set_option( 'stopwatch_enabled', gpc_get_int( 'stopwatch_enabled', OFF ) );
+$config = \TimeTrackingPlugin::getConfig();
+foreach($config as $opt => $default ) {
+
+    if( is_int($default) ) {
+        $fn = 'gpc_get_int';        
+        if( strpos($opt, 'enabled') !== FALSE ) {
+            // is a bool like 1/0
+            $default = OFF;
+        }
+    }
 
-form_security_purge( 'plugin_TimeTracking_config_update' );
+    if( is_string($default) ) {
+        $fn = 'gpc_get_string';        
+    }
+    plugin_config_set( $opt, $fn( $opt, $default ) );        
```

Comment:
This is not clean in the way that it will always save the option in database, even if the value is the same as the default.
That was the intention of the previous function `maybe_set_option()`. Actually that function may be also wrong, as it really should do:
1) if the new value is different from default, save it
2) if the new value is same as default: don't save it, and delete the current value if it exists.

That's the usual way mantis treats configuration options

Reply:
I can rethink this, but it's not clear for me why this so important to avoid a write if value has not been changed. If standard way to work has to be as described, then IMHO it will be better that plugin_config_set() will do all the logic. What do you think about this last approach?

Reply:
Reviewing the working of plugin config, it's a bit different from core config.
The plugin config options are always stored in database, as part of plugin setup, which reads the initial values from calling `config()`
So, it's not bad to "always write" the option. Not clean either... but..ok

_Note, the scnario i described was for core configurations, where values may exists as defaults in configuration file instead of database_

>  important to avoid a write if value has not been changed

one reason is to avoid unneeded database writes.